Example/hierarchy (#4923)
payloadcms/payload: examples/hierarchy/README.md

# Payload Hierarchy Example

> A hierarchy for [Payload](https://github.com/payloadcms/payload).

Reviewer: ```suggestion
This example demonstrates how to achieve a virtual hierarchy between documents in your [Payload](https://github.com/payloadcms/payload) application.```

> This repo was created by running `npx create-payload-app@latest` and selecting the "blank" template then it was modified.

Reviewer: ```suggestion
```

> See the official [Examples Directory](https://github.com/payloadcms/payload/tree/main/examples) for details on how to use Payload in a variety of different ways.

Reviewer: ```suggestion
```

## Quick Start

To spin up the project locally, follow these steps:

1. First clone the repo
1. Then `cd YOUR_PROJECT_REPO && cp .env.example .env`
1. Next `yarn && yarn dev` (or `docker-compose up`, see [Docker](#docker))
1. Now `open http://localhost:3000/admin` to access the admin panel
1. Create your first admin user using the form on the page

That's it! Changes made in `./src` will be reflected in your app.

## How it works

> This is an example parent-child hierarchy between Entities and People.  Entities use of a virtual field to find people and entities it is related to.

Reviewer: This example achieves parent/child relationships between your documents through the use of virtual fields. When you query a document with the `?children=true` query param, an `afterRead` hook is used to populate the documents within its own tree.

For more information on how virtual fields, see the [Official Virtual Fields Example]([Virtual Fields](https://github.com/payloadcms/payload/tree/main/examples/virtual-fields).
